### ISSUE TITLE
why predict result is different in same model and environment?

### ISSUE THREAD

**Issue by Helloworld0214** [wenwenyu/PICK-pytorch#90]
Thank for your excellent work. I use PICK to train invoices, but why the predict result is different in same model and environment? Looking forward to your reply.

**Comment by Helloworld0214** [comment]
I got the answer. The test process need add random seed. @wenwenyu

**Comment by jorgerodriguezsj** [comment]
@Helloworld0214 Where do you set the random seed?

**Comment by Helloworld0214** [comment]
> @Helloworld0214 Where do you set the random seed?

I set it before in  test model  loaded.
![image](https://user-images.githubusercontent.com/22692122/118914986-c1895780-b95e-11eb-8d03-c2197fa2b9a1.png)

**Comment by IrvingHerreraLuna** [comment]
I have the same problem, I trained with GPU on an EC2 instance, now I am running it on CPU and the test results change, Can I be more explicit in the use of the seed? This is during the test? Even inserting the seed the result is different

**Comment by Helloworld0214** [comment]
> I have the same problem, I trained with GPU on an EC2 instance, now I am running it on CPU and the test results change, Can I be more explicit in the use of the seed? This is during the test? Even inserting the seed the result is different

yes, I modify code in test file.
```python
SEED = 2021
def set_seed():
    torch.manual_seed(SEED)
    torch.cuda.manual_seed(SEED)
    torch.backends.cudnn.benchmark = False
    torch.backends.cudnn.deterministic = True
    np.random.seed(SEED)
    random.seed(SEED)
```
and model_eval function in parse_config
`
def model_eval(self):
    module_args = self._config['model_arch']['args']
    # print(dict(module_args))
    for Key, Value in module_args.items():
        if isinstance(Value, collections.OrderedDict):
            for key, value in Value.items():
                if key == 'dropout':
                    Value[key] = 0.0
                elif isinstance(value, collections.OrderedDict):
                    for k, v in value.items():
                        if k == 'dropout':
                            value[k] = 0.0
`

**Comment by IrvingHerreraLuna** [comment]
config refers to the class or the instance within the ConfigParser class?
Thanks for your quick response

**Comment by ImmineoII** [comment]
Hi, can you post the set_seed() function? @Helloworld0214
